### PR TITLE
Full incremental objects

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

**Comment by jdoliner** [comment]
@derekchiang whenever you have a chance this is in need of review

**Comment by derekchiang** [comment]
@jdoliner can you summarize what this PR does?

**Comment by jdoliner** [comment]
Sure, this pr adds a few extra features to our API. The main one is that pps rather than incremental being a boolean it's become an enum with 3 values `NONE` `DIFF` and `FULL`. `NONE` and `DIFF` correspond to `true` and `false` before. `FULL` is a new behavior in which files that have nonzero diffs are shown in their entirety.

This also required adding a `FullFile` flag to file operations in the PFS api which means that you want to see FullFiles if there's any diff. It's part of a new `DiffMethod` object.

**Comment by derekchiang** [comment]
LGTM, other than maybe adding some comment to the protobuf file explaining what `fullFile` and `DiffMethod` do.
